### DISCUSSION TITLE
Please fix the API approval process

### DISCUSSION THREAD

**Discussion by littlebee** [etsy/open-api#1134] [Ideas]
I get it.  It's tough keeping bad actors out, but seriously, how about just a little parlay.   

You apply for a key, but there is no e-mail, notification, Etsy message,  or status updates, so you need to check back every day at (https://www.etsy.com/developers/your-apps) to see if it's been approved and after waiting over a week, one day it just says it's rejected because "Missing valid website URL".  I gave it my company domain www.ecosfab.com which right now is just redirecting to Etsy root because my shop will be my web presence to start out, but my shop isn't ready yet and when it is I'll change the redirect.   But how do I know that is why it's "invalid"??  I don't.  Where / who would I ask?? IDK.

**Comment by littlebee** [comment]
Also, I should have mentioned, there was a brief period between "pending" and "rejected" where there were no apps showing on .../your-apps.   Which in turn caused me to submit a new application for the API key that probably has the same "invalid" website as the first.   Maybe allow people to correct an application or remove and add another if they realize they made a mistake?  Please.

**Comment by simonj** [comment]
You need to create a landing page for them to see what your app is about if it's commercial. Also had to do that.

And they don't have any test API it's only LIVE data you are working with.

**Reply by brettcthompson25** [reply]
What does this landing page need to consist of? My application is going to be entirely local and backend, do I still need to create a website and a landing page?

**Reply by littlebee** [reply]
Ignore my deleted comments.  I thought I was on a different thread about API authentication.   

The answer is that they will accept your Etsy store URL (like https://www.etsy.com/shop/Ecosfab), but using the Etsy root URL (https://etsy.com) is unacceptable.

**Comment by littlebee** [comment]
Thank you.  Yeah, I was guessing that was it.  I changed the domain redirect to point to my shop.  Hopefully that’s enough to reinforce what I wrote in the application - this is not an app intended to be used on any shop but mine.
